### DISCUSSION TITLE
How to evaluate second-order phase transition with free energy hessian?

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#370] [General]
Hi, I find that there are two ways to minimize the free energy.
1, minimize free energy with respect to auxiliary FCs **$\Phi$** and centroids **_R_**
2, for a given **_R_**, minimize free energy with respect to only auxiliary FCs **$\Phi$**

I believe that the first way is to access the converged free energy at the specified temperature. For example, in the case of second-order phase transition, if I set the T < Tc (critical temperature), the final **_R_** should be low-symmetry phase **$R_{ls}$**. In this case, I think the free energy hessian with respect to **$R_{ls}$** is also positive-difinite.

I find that SSCHA evaluates second-order phase transition by the second minimization way. 
Is it as stated below?
If I set the given **_R_** to high-symmetry phase **$R_{hs}$** and do minimization with respect to only **$\Phi$**, I can use free energy to calculate the hessian with respect to **$R_{hs}$**.
And this hessian is positive-difinite at T > Tc, has at least one zero eigenvalue at T = Tc, and has negative eigenvalues at T < Tc. Then I can evaluate the second-order phase transition.

**Comment by mesonepigreco** [comment]
Yes, this is correct. However, if you start from a high-symmetry configuration, for symmetry reasons, even if you relax centroids and FCs, you cannot escape from the high-symmetry configurations, so in practice, you often relax everything together. 
Also, for the Hessian to have a physical meaning, you must have the gradient of the free energy concerning the centroid equal to zero, so you must always relax both centroids and FCs. Indeed, since the code preserves the symmetries in minimization, you are not ensured to reach the global minimum of the free energy landscape.

**Reply by biglinn** [reply]
Indeed, I have a doubt.
With appropriate initial centorids and FCs, I do SSCHA at a range of temperatures. The SSCHA output centroids and FCs at each temperature have a physical meaning. For example, at **T < Tc**, the centroids should be $R_{ls}$ which is equilibrium configuration at this temperature.

In this case, the free energy is the real free energy at this temperature(T < Tc). Does the hessian have negative eigenvalues?
Since the free energy is real at this temperature, shouldn't the hessian be positive-definite?
How do I evaluate second-order phase transition?

**Reply by mesonepigreco** [reply]
Yes, the hessian is positive-definite in that case. There is an example in the paper of the code: https://iopscience.iop.org/article/10.1088/1361-648X/ac066b/pdf

Figure 3, panel d, you find the Hessian eigenvalues as a function of temperature both in the low symmetry phase and in the high-symmetry phase. Indeed, only those of the high symmetry phase (Fm-3m in that case) become negative, while those of the high-symmetry one (R3m) remain always positive, except exactly at the critical point where they should reach zero. 

That case is slightly more complicated as it is a first-order phase transition, so you have two critical points corresponding to the upper and lower transition temperature, but the concept is the same

**Reply by biglinn** [reply]
Thanks! I'll carefully read this article!

**Comment by biglinn** [comment]
Hi, @mesonepigreco 

I have some ideas about the paper recommended to me. Could you verify if they are correct?
<img width="532" alt="hessian" src="https://github.com/user-attachments/assets/ace88139-4f97-44b5-b758-fae8126e74c6" />

What I understand is that $D^{(S)}$ is given by $\Phi_{eq}$ and is **always positive-definite**. So $D^{(S)}$ couldn't be used to evaluate phase transition.
But $D^{(S)}$ is part of $D^{(F)}$. We use eigenvalues of $D^{(F)}$ to evaluate phase transition.

**Reply by mesonepigreco** [reply]
Exactly

**Comment by biglinn** [comment]
Hi, @mesonepigreco 

I have something that confuses me.
I give an example at T < Tc. As follows:
<img width="592" alt="hessian_1" src="https://github.com/user-attachments/assets/bcb625f3-5b59-4142-949a-0417de8bda7a" />

At T < Tc, the SSCHA $R_{eq}$ should be $R_{ls}$, but the hessian with respect to $R_{ls}$ is always positive-definite according to the free energy curvature.
In this case of T< Tc, the hessian with respect to $R_{hs}$ have negative eigenvalues, which could represent the disability.

So I think that the hessian should be always with respect to $R_{hs}$, whatever T is.  It is right to think so?

**Reply by mesonepigreco** [reply]
Yes, if you want to compute the transition temperature, that is the right thing to do; use always the high-symmetry phase (so you exploit the higher number of symmetries to reduce the computational cost of the SSCHA calculation) and check the point at which the free energy hessian becomes imaginary. 

If you explore the low symmetry phase instead, you get the point where your system acquires more symmetries (the critical temperature). If you compute the free energy Hessian, you will always get positive definite phonons, but they go to 0 at the transition temperatures from both low and high temperatures.

**Reply by biglinn** [reply]
Technically, using the high-symmetry phase refers to use harmonic dynamical matrix of high-symmetry configuration as initial guess. Is it right?

Moreover, I find that the structure and dyn obtained at the previous lower temperature is used as input of this temperature (from https://sscha.eu/Tutorials/tutorial_03_secondorder_phase_transitions/ and the paper that you mentioned _J. Phys.: Condens. Matter 33 363001_).
If I use harmonic dyn of high-symmetry configuration as initial guess to start, should I start from high temperature to low temperture?
Or, could I simply do SSCHA at different temperatures from harmonic dyn? (in other words, not use previous output as input)
